app/models/podcast_import.rb

```diff
@@ -0,0 +1,376 @@
+
+  attr_accessor :feed, :feed_raw_doc, :templates
+
+  belongs_to :podcast, -> { with_deleted }, touch: true, optional: true, autosave: true
+  has_many :episode_imports, -> { where(has_duplicate_guid: false).includes(:podcast_import) }, dependent: :destroy
+
+  before_validation :set_defaults, on: :create
+
+  validates :account_id, :url, presence: true
+
+  COMPLETE = "complete".freeze
+  FAILED = "failed".freeze
+
+  CREATED = "created".freeze
+  STARTED = "started".freeze
+  FEED_RETRIEVED = "feed retrieved".freeze
+  RETRYING = "retrying".freeze
+  IMPORTING = "importing".freeze
+  PODCAST_CREATED = "podcast created".freeze
+
+  def episode_import_placeholders
+    EpisodeImport.where(podcast_import_id: id).having_duplicate_guids
+  end
+
+  def config_url=(config_url)
+    c_url = Addressable::URI.parse(config_url)
+    response = connection(c_url).get(c_url.path, c_url.query_values)
+    self.config = ActiveSupport::JSON.decode(response.body).try(:with_indifferent_access)
+  end
+
+  def set_defaults
+    self.status ||= CREATED
+    self.config ||= {
+      audio: {},            # map of guids to array of audio file urls
+      episodes_only: false  # indicates if podcast should be updated
+    }
+  end
+
+  def episode_importing_count
+    feed_episode_count - episode_import_placeholders.count
+  end
+
+  def status
+    return super unless episode_imports.count > 0
+    return super if episode_importing_count > episode_imports.count
+
+    if complete?
+      COMPLETE
+    elsif finished? && some_failed?
+      FAILED
+    else
+      super
+    end
+  end
+
+  def finished?
+    return false unless episode_imports.count == episode_importing_count
+    episode_imports.all? do |e|
+      e.status == EpisodeImport::COMPLETE ||
+        e.status == EpisodeImport::FAILED
+    end
+  end
+
+  def complete?
+    return false unless episode_imports.count == episode_importing_count
+    episode_imports.all? { |e| e.status == EpisodeImport::COMPLETE }
+  end
+
+  def some_failed?
+    episode_imports.any? { |e| e.status == EpisodeImport::FAILED }
+  end
+
+  def retry!
+    update(status: RETRYING)
+    import_later
+  end
+
+  def import_later(import_podcast = true)
+    PodcastImportJob.perform_later(self, import_podcast)
+  end
+
+  def import_podcast!
+    update!(status: STARTED)
+
+    # Request the RSS feed
+    get_feed
+    update!(status: FEED_RETRIEVED)
+
+    # Create the podcast
+    create_or_update_podcast!
+
+    update!(status: PODCAST_CREATED)
+  rescue => err
+    update(status: FAILED)
+    raise err
+  end
+
+  def import_episodes!
+    # Request the RSS feed again
+    get_feed
+
+    # Update podcast attributes
+    create_or_update_podcast!
+    update!(status: PODCAST_CREATED)
+
+    # Create the episodes
+    update!(status: IMPORTING)
+    create_or_update_episode_imports!
+  rescue => err
+    Rails.logger.error ([err.message] + err.backtrace).join($/)
+    update(status: FAILED)
+    raise err
```

Comment:
In this case, it might make sense to think about retries.  If the status is actually `FAILED` then it's not immediately clear to me that the job should be retried.  These are currently idempotent , so it will do no harm.